src/gui/socketapi.cpp

```diff
@@ -491,23 +491,72 @@ void SocketApi::command_SHARE_MENU_TITLE(const QString &, SocketListener *listen
     listener->sendMessage(QLatin1String("SHARE_MENU_TITLE:") + tr("Share with %1", "parameter is ownCloud").arg(Theme::instance()->appNameGUI()));
 }
 
-void SocketApi::command_COPY_PRIVATE_LINK(const QString &localFile, SocketListener *)
+// Fetches the private link url asynchronously and then calls the target slot
+template <typename T>
+void fetchPrivateLinkUrl(const QString &localFile, typename QtPrivate::FunctionPointer<T>::Object *target,
```

Comment:
Don't use QtPrivate,  this is not Qt public API and it might be changed.
Just use SocketApi* or another template type. 
(maybe no need to have a template:  `void fetchPrivateLinkUrl(const QString &localFile, SocketApi *target, void (SocketApi::*targetFunc)() ) `